### DISCUSSION TITLE
(NextJS) Error: Please call "createAppKit" before using "useAppKit" hook

### DISCUSSION THREAD

**Discussion by Rukny** [WalletConnect/walletconnect-monorepo#5435] [Developer Support] [answered]
(Next.js)
Link to Repo:
https://github.com/Rukny/reown-appkit-test

I already have createAppKit defined correctly in app/providers and have that component wrapped correctly in layout.tsx
![image](https://github.com/user-attachments/assets/de610d5e-99ab-4584-bc85-9e1e7f6c65b6),

However when I try to call the useAppKit() hook in this client component
![image](https://github.com/user-attachments/assets/df269825-7c80-4765-afd0-a243e6eda521)
I get this error at runtime
![image](https://github.com/user-attachments/assets/cccd7bdb-63fe-4721-b001-a8e91e84e170)

If I use the w3m-button component it works with all the configurations that I defined in /providers

**Comment by andriy-shymkiv** [comment]
Faced with the same issue today

**Comment by Milordiksepol** [comment]
bob

**Comment by kyruzic** [comment, marked as answer]
The issue is where you are importing createAppKit from, you need to use

`import { createAppKit } from '@reown/appkit/react';`

NOT 

`import { createAppKit } from '@reown/appkit';`

**Reply by lovelgeorge99** [reply]
Yes this works for me

**Reply by Rukny** [reply]
Thank you

**Reply by nimishjn** [reply]
I am importing it from the correct place i.e. `@reown/appkit/react`, and yet I am facing this issue

**Reply by vladislavbairash** [reply]
Yes, it's correct solution

**Comment by TornGuard** [comment]
Facing this exact same issue today. According to the default next js setup docs here  https://docs.reown.com/appkit/next/core/installation.  

How is the modal function used on the context/index because eslint shows that modal is declared but never read? I think it could be the reason for the error.

**Reply by powxenv** [reply]
> Facing this exact same issue today. According to the default next js setup docs here https://docs.reown.com/appkit/next/core/installation.
> 
> How is the modal function used on the context/index because eslint shows that modal is declared but never read? I think it could be the reason for the error.

make sure you’re using "use client" and not a server component

**Reply by nimishjn** [reply]
This fixed the issue. Thank you!

**Comment by InfiniteAengus** [comment]
```js
'use client';

import { Button } from '@/components/ui/button';
import { useAppKit } from '@reown/appkit/react';

export default function Home() {
  const { open } = useAppKit();

  return (
    <div className="grid min-h-screen grid-rows-[20px_1fr_20px] items-center justify-items-center gap-16 p-8 pb-20 font-[family-name:var(--font-geist-sans)] sm:p-20">
      <Button onClick={() => open()}>Connect</Button>
    </div>
  );
}
```

I've used all of that, but returns same error

**Reply by vedansh-pt** [reply]
found any solution?

**Comment by UN-TrqIezTocLrnEDr** [comment]
When importing `createAppKit`, it was initially set as `@reown/appkit`(this was automatically suggested by the IDE). However, changing it to `@reown/appkit/react` resolved the issue.

The docs show it this way too.
